### PR TITLE
fix: wallet filtering in a non-exclusive way

### PR DESCRIPTION
**Pull request by markettes**

### CODE REVIEW

File: src/Pages/Wallets.razor [hunk 1]
```diff
@@ -112,26 +114,26 @@
...
-                                    <Check TValue="bool" @bind-Checked="_archivedWalletFilter">📂 Archived</Check>
-                                    <Check TValue="bool" @bind-Checked="_compromisedWalletFilter">⛔️ Compromised</Check>
+                                    <Check TValue="bool"  CheckedChanged="@(() => { _hotWalletFilter = !_hotWalletFilter; OnCheckedChanged(true); })">🔥 Hot Wallet</Check>
+                                    <Check TValue="bool"  CheckedChanged="@(() => { _coldWalletFilter = !_coldWalletFilter; OnCheckedChanged(true); })">❄️ Cold Wallet</Check>
+                                    <Check TValue="bool"  CheckedChanged="@(() => { _finalisedWalletFilter = !_finalisedWalletFilter; OnCheckedChanged(true); })">🏁 finalised</Check>
+                                    <Check TValue="bool"  CheckedChanged="@(() => { _archivedWalletFilter = !_archivedWalletFilter; OnCheckedChanged(true); })">📂 Archived</Check>
+                                    <Check TValue="bool"  CheckedChanged="@(() => { _compromisedWalletFilter = !_compromisedWalletFilter; OnCheckedChanged(true); })">⛔️ Compromised</Check>
```
Comment (RodriFS) [inline src/Pages/Wallets.razor:136]: You could create a function for not repeating so much code. And I don't know how dotnet handles this, but most probably every time the component rerenders, new lambda functions are created for every Check tag

File: src/Pages/Wallets.razor [hunk 2]
```diff
@@ -140,7 +142,7 @@
                         <span>
                             Wallet Properties
                         </span>
-                        <span title="@("🔥 Hot Wallet\n❄️ Cold Wallet\n🏁 Finalized\n📂 Archived\n⛔️ Compromised")">
+                        <span title="@("🔥 Hot Wallet\n❄️ Cold Wallet\n🏁 Finalised\n📂 Archived\n⛔️ Compromised")">
```
Comment (RodriFS) [inline src/Pages/Wallets.razor:145]: 🇬🇧💯

File: src/Pages/Wallets.razor [hunk 3]
```diff
@@ -1237,17 +1241,28 @@
...
+        if (_hotWalletFilter && wallet.IsHotWallet)
+            return true;
+
+        if (_coldWalletFilter && !wallet.IsHotWallet)
+            return true;
+
+        if (_finalisedWalletFilter && wallet.IsFinalised)
+            return true;
+
+        return false;
```
Comment (RodriFS) [inline src/Pages/Wallets.razor:1265]: you can replace the last bit with the following, so it's more compact:

```
return _archivedWalletFilter && wallet.IsArchived
        || _compromisedWalletFilter && wallet.IsCompromised
        || _hotWalletFilter && wallet.IsHotWallet
        || _coldWalletFilter && !wallet.IsHotWallet
        || _finalisedWalletFilter && wallet.IsFinalised;
```

**Review by RodriFS — APPROVED** [review]

File: src/Pages/Wallets.razor (hunk 2 above)
Comment (markettes) [reply src/Pages/Wallets.razor:145]: 😂

**Review by RodriFS — APPROVED** [review]

**Pull request merged**